Editor/Utils/ScanSolidTexture.cs

```diff
@@ -0,0 +1,214 @@
+                    RenderTextureFormat.Default,
+                    RenderTextureReadWrite.sRGB);
+
+
+                // Backup the currently set RenderTexture
+                RenderTexture previous = RenderTexture.active;
+                Graphics.Blit(texture2D, tmp);
+                RenderTexture.active = tmp;
+
+                Texture2D myTexture2D = new Texture2D(texture2D.width, texture2D.height);
+                myTexture2D.ReadPixels(new Rect(0, 0, tmp.width, tmp.height), 0, 0);
+                //Graphics.CopyTexture(texture2D, myTexture2D);
+                //var pixel = myTexture2D.GetPixel(4, 0);
+                myTexture2D.Apply();
+
+                if (IsSolidColor(myTexture2D))
+                {
+                    if (texture2D.width != 1 && texture2D.height != 1)
+                    {
+                        isTooBig = true;
+                    }
+                }
+
+                RenderTexture.active = previous;
+                RenderTexture.ReleaseTemporary(tmp);
+            }
+
+            return isTooBig;
+        }
+
+        /// <summary>
+        /// Check if a texture is comprised of a single solid color.
+        /// </summary>
+        /// <param name="texture">The texture to check.</param>
+        /// <returns>True if the texture is a single solid color.</returns>
+        static bool IsSolidColor(Texture2D texture)
+        {
+            // Skip "degenerate" textures like font atlases
+            if (texture.width == 0 || texture.height == 0)
+            {
+                return false;
+            }
+
+            var pixels = texture.GetPixels32();
+
+            // It is unlikely to get a null pixels array, but we should check just in case
+            if (pixels == null)
+            {
+                Debug.LogWarning($"Could not read {texture}");
+                return false;
+            }
+
+            // It is unlikely, but possible that we got this far and there are no pixels.
+            var pixelCount = pixels.Length;
+            if (pixelCount == 0)
+            {
+                Debug.LogWarning($"No pixels in {texture}");
+                return false;
+            }
+
+            // Convert to int for faster comparison
+            var colorValue = Color32ToInt.Convert(pixels[0]);
+            var isSolidColor = true;
+            for (var i = 1; i < pixelCount; i++)
+            {
+                var pixel = Color32ToInt.Convert(pixels[i]);
+                if (pixel != colorValue)
+                {
+                    isSolidColor = false;
+                    break;
+                }
+            }
+
+            return isSolidColor;
+        }
+    }
+
+    /// <summary>
+    /// Conversion struct which takes advantage of Color32 struct layout for fast conversion to and from Int32.
+    /// </summary>
+    /// <author>
+    /// Unity Labs SuperScience
+    /// <author>
+    [StructLayout(LayoutKind.Explicit)]
+    public struct Color32ToInt
```

Comment:
That's an interesting struct and optimization!

If there is a chance we may use this again I would say it could move to a utility file like some other of our utilities.

For example:  "Utils/TextureUtils.cs" or so?

Reply:
EDIT:  Oh, I just noticed that this is inside a "Utils" folder file. :)

Maybe we just rename the file to "TextureUtils.cs", and maybe there is a chance we move more code here.

So it is not so specific to "Scan Solid Texture" basically.